### ISSUE TITLE
Nim contracts on eWASM

### ISSUE THREAD

**Issue by kggp1995** [ewasm/testnet#122]
**Objective:** Develop a detailed pipeline for Contracts in Nim based on the eWASM Specs

**People involved:** @kggp1995, @AchalaSB and others from eWASM & status.im team

**Goal(s):**
1. Develop a proper emscripten-based transpiler that can convert a `.nim` program (contract) to WASM executable along with other supporting files

**Detailed Workflow:**
WIP - Coming up soon..

**Comment by kggp1995** [comment]
@achalaSB can explain the processes in detail, perhaps?

Also if @lrettig , @axic or someone can confirm the flow's alignment - that'll be great.

**Comment by axic** [comment]
I'm sure you are aware of https://github.com/status-im/nim-eth-contracts ? It does some of these steps already.

**Comment by kggp1995** [comment]
Yes, but unfortunately it does not support an emscripten-based pipeline, yet.

**Comment by kggp1995** [comment]
@AchalaSB , can you describe the pipeline for Nim Contract in detail?

That should help here. TIA!

**Comment by AchalaSB** [comment]
@axic I have compiled the Nim contract using `emscripten` but the generated `wasm` file is not according to the eWASM spec, couldn't deploy in testnet.
How can I get the required `wasm` file?

**Comment by axic** [comment]
Does it have the proper imports and exports?

You could run wasm-chisel on it to clean some of that up. You need to install it via Rust's cargo: `cargo install chisel`.
